### PR TITLE
Order_by_button

### PR DESCRIPTION
**Pull request by saundlj**
- Added order by button for feed page (styling still needs to be done). Once working and a suitable class is made, we need to add this button to View Job Posting and Job Applications. 

- Addition unit tests for remaining forms

- Validation for remaining forms in contollers.py

- Meta data more readable in MyPosts 

- added relationship for application to post

- additional wtform validation in FeedApplyForm

### CODE REVIEW

**Comment by vittopan** [comment]
lovely stuff.

File: instance/memory
Comment (tomve1tch) [inline instance/memory:1]: this should not be commited

File: app/__init__.py [hunk 1]
```diff
@@ -31,11 +31,11 @@ def create_app(config=DeploymentConfig):
...
-        if config == DeploymentConfig:
-            from test_data import deployment_data
-            db.drop_all()
-            db.create_all()
-            deployment_data()
+        # if config == DeploymentConfig:
+        #     from test_data import deployment_data
+        #     db.drop_all()
+        #     db.create_all()
+        #     deployment_data()
```
Comment (tomve1tch) [inline app/__init__.py:38]: correctly commenting out to prevent db resetting at each flask run

File: app/controllers.py [hunk 2]
```diff
@@ -161,6 +161,22 @@ def new_bio(bio:Account):
...
+def new_application(application:Application, post:Post):
+
+    # Cover Letter check 
+    application.cover_letter, invalid_string = sanitize_input(application.cover_letter, description=True)
+    if invalid_string:
+        raise ApplicationFormError("Cover Letter contains at least one invalid character. Please remove and try again.")
+    
+    # cannot apply for own job post
+    if post.user_id == application.user_id:
+        raise ApplicationFormError("Cannot apply for your own job post!")
```
Comment (tomve1tch) [inline app/controllers.py:176]: good use of data validation

File: app/forms.py [hunk 3]
```diff
@@ -90,12 +90,7 @@ class UserAccountForm(FlaskForm):
     submit = SubmitField("Submit")
 
 class FeedApplyForm(FlaskForm):
-    cover_letter = TextAreaField('Cover Letter', validators=[DataRequired()])
-    post_id = IntegerField('Post ID')
-    submit = SubmitField("Submit")
-
-class FeedApplyForm(FlaskForm):
-    cover_letter = TextAreaField('Cover Letter', validators=[DataRequired()])
+    cover_letter = TextAreaField('Cover Letter', validators=[DataRequired(), Length(min=10, max=500, message='Cover Letter must be between 10 and 500 characters.')])
```
Comment (tomve1tch) [inline app/forms.py:93]: again, good data validation

File: app/routes.py [hunk 4]
```diff
@@ -101,10 +102,17 @@ def feed():
         if max_salary is not None:
             query = query.filter(Post.salary <= max_salary)
 
-        job_posts = query.all()
+        
+        if sort_order == 'asc':
+            job_posts = query.order_by(Post.date_posted.asc()).all()
+        else:
+            job_posts = query.order_by(Post.date_posted.desc()).all()
```
Comment (tomve1tch) [inline app/routes.py:109]: nice feature, looks good

**Review by tomve1tch — APPROVED** [review]
Looks good mate, will delete memory from git after pull

**Pull request merged**